Source/FieldSolver/SpectralSolver/SpectralAlgorithms/PsatdAlgorithmFirstOrder.cpp

```diff
@@ -0,0 +1,368 @@
+                                     + C12*Jx_c1 + C13*Jy_c1 + C14*Jz_c1 // only with J linear in time
+                                     + C15*rho_c0  // only with div cleaning
+                                     + C16*rho_c1; // only with div cleaning and rho linear in time
+
+                // Ez
+                C01 = (div_cleaning) ? 0._rt : kx*kz*(1._rt-C)/knorm2;
+                C02 = (div_cleaning) ? 0._rt : ky*kz*(1._rt-C)/knorm2;
+                C03 = (div_cleaning) ? C : (kx2*C+ky2*C+kz2)/knorm2;
+                C04 = -I*c*ky*S/knorm;
+                C05 = I*c*kx*S/knorm;
+                C06 = 0._rt;
+                C07 = (div_cleaning) ? I*c*kz*S/knorm : 0._rt;
+                C09 = (div_cleaning) ? 0._rt : mu0*c*kx*kz*(knorm*S-dt*c*knorm2)/knorm4;
+                C10 = (div_cleaning) ? 0._rt : mu0*c*ky*kz*(knorm*S-dt*c*knorm2)/knorm4;
+                C11 = (div_cleaning) ? -mu0*c*S/knorm : -mu0*c*(dt*c*kz2*knorm2+kx2*knorm*S+ky2*knorm*S)/knorm4;
+                C12 = (J_linear && !div_cleaning) ? mu0*kx*kz*(2._rt*(1._rt-C)-dt2*c2*knorm2)/(2._rt*knorm4) : 0._rt;
+                C13 = (J_linear && !div_cleaning) ? mu0*ky*kz*(2._rt*(1._rt-C)-dt2*c2*knorm2)/(2._rt*knorm4) : 0._rt;
+                C14 = 0._rt; // This is not redundant, do not remove this
+                if (J_linear) C14 = (div_cleaning) ? mu0*(C-1._rt)/knorm2 : mu0*(2._rt*kx2*(C-1._rt)+2._rt*ky2*(C-1._rt)-dt2*c2*kz2*knorm2)/(2._rt*knorm4);
+                C15 = (div_cleaning) ? I*mu0*c2*kz*(C-1._rt)/knorm2 : 0._rt;
+                C16 = (div_cleaning && rho_linear) ? I*mu0*c*kz*(knorm*S-dt*c*knorm2)/knorm4 : 0._rt;
+
+                fields(i,j,k,Idx.Ez) = C01*Ex_old + C02*Ey_old + C03*Ez_old
+                                     + C04*Bx_old + C05*By_old + C06*Bz_old
+                                     + C07*F_old // only with div cleaning
+                                     + C09*Jx_c0 + C10*Jy_c0 + C11*Jz_c0
+                                     + C12*Jx_c1 + C13*Jy_c1 + C14*Jz_c1 // only with J linear in time
+                                     + C15*rho_c0  // only with div cleaning
+                                     + C16*rho_c1; // only with div cleaning and rho linear in time
+
+                // Bx
+                C01 = 0._rt;
+                C02 = I*kz*S/(c*knorm);
+                C03 = -I*ky*S/(c*knorm);
+                C04 = (div_cleaning) ? C : (kx2+ky2*C+kz2*C)/knorm2;
+                C05 = (div_cleaning) ? 0._rt : kx*ky*(1._rt-C)/knorm2;
+                C06 = (div_cleaning) ? 0._rt : kx*kz*(1._rt-C)/knorm2;
+                C08 = (div_cleaning) ? I*kx*S/(c*knorm) : 0._rt;
+                C09 = 0._rt;
+                C10 = I*mu0*kz*(C-1._rt)/knorm2;
+                C11 = -I*mu0*ky*(C-1._rt)/knorm2;
+                C12 = 0._rt;
+                C13 = (J_linear) ? I*mu0*kz*(knorm*S-dt*c*knorm2)/(c*knorm4) : 0._rt;
+                C14 = (J_linear) ? -I*mu0*ky*(knorm*S-dt*c*knorm2)/(c*knorm4) : 0._rt;
```

Comment:
We could also define `1/c` as a temporary variable.

Reply:
Done in 6fe0880.